src/libsystemd/sd-bus/sd-bus.c

```diff
@@ -3335,7 +3339,11 @@ _public_ int sd_bus_wait(sd_bus *bus, uint64_t timeout_usec) {
         if (bus->rqueue_size > 0)
                 return 0;
 
-        return bus_poll(bus, false, timeout_usec);
+        r = bus_poll(bus, false, timeout_usec);
+        if (r < 0 && ERRNO_IS_TRANSIENT(r))
+                return 1; /* treat EINTR as success, but let's exit, so that the caller will call back into us soon. */
```

Comment:
isn't this an API change?

Reply:
no. 

returning 0 means timeout reached. > 0 means "something happened, please call sd_bus_process()". And that's exactly what we are fine with here. Of course, since no IO actually happened sd_bus_process() will then be a NOP, but that's totally OK. The caller will then have to call sd_bus_wait() again, which is exactly what is intended here.

Or in other words: we take the liberty to return 1 any time we like. Given that the caller doesn't get access to the IO events directly anyway (they only think they can do is call sd_bus_process() to get them handled), this is perfectly compatible.